kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesMockServer.java

```diff
@@ -88,12 +89,21 @@ public RecordedRequest getLastRequest() throws InterruptedException {
     }
 
     public NamespacedKubernetesClient createClient() {
+        Config config = getMockConfiguration();
+        return new DefaultKubernetesClient(createHttpClientForMockServer(config), config);
+    }
+
+    protected Config getMockConfiguration() {
         Config config = new ConfigBuilder()
                 .withMasterUrl(url("/"))
                 .withTrustCerts(true)
                 .withTlsVersions(TLS_1_2)
                 .withNamespace("test")
                 .build();
-        return new DefaultKubernetesClient(createHttpClientForMockServer(config), config);
+        config.setCurrentContext(null);
+        config.setContexts(Collections.emptyList());
+        config.setOauthToken(null);
+
+        return config;
     }
```

Comment:
Why isn't this working?
```java
        Config config = new ConfigBuilder()
                .withMasterUrl(url("/"))
                .withTrustCerts(true)
                .withTlsVersions(TLS_1_2)
                .withNamespace("test")
                .withContexts(Collections.emptyList())
                .build();
        return new DefaultKubernetesClient(createHttpClientForMockServer(config), config);
```

Reply:
`ConfigBuilder()` calls this:
```java
    public ConfigBuilder() {
        this(true);
    }

    public ConfigBuilder(java.lang.Boolean validationEnabled) {
        this(new Config(), validationEnabled);
    }
```

`Config()` enables autoConfigure
```java
  public Config() {
    this(!Utils.getSystemPropertyOrEnvVar(KUBERNETES_DISABLE_AUTO_CONFIG_SYSTEM_PROPERTY, false));
  }
```
let me try to use `ConfigBuilder(Config.empty())` instead

Reply:
I'm not following, in your changes you are using `new ConfigBuilder()` anyway.
The explicit calls to the setters (`setCurrentContext`, etc.) with null values should be redundant as (hopefully) those variables should be initialized as null.